lib/matplotlib/text.py

```diff
@@ -2092,7 +2093,9 @@ def draw(self, renderer):
                 self.arrow_patch.figure = self.figure
             self.arrow_patch.draw(renderer)
 
-        Text.draw(self, renderer)
+        astext = copy.copy(self)
+        astext.__class__ = Text
```

Comment:
Well, if that is the case, then I am -1 on this kludge. We know exactly
what the problem is -- we shouldn't be using bbox_artist() for this.
Perhaps "bbox_text()"? It would work similarly to bbox_artist, but would
use the Text class's get_window_extent no matter what object is passed to
it?

As for precedence for explicitly changing an object's __class__, I am
assuming that you are referring to mplot3d. Yeah, let's not bring that wart
into matplotlib proper. I still haven't figured out a way to fix that.


On Mon, Feb 23, 2015 at 4:58 PM, Cimarron <notifications@github.com> wrote:

> In lib/matplotlib/text.py
> <https://github.com/matplotlib/matplotlib/pull/4145#discussion_r25206783>:
>
> > @@ -2092,7 +2093,9 @@ def draw(self, renderer):
> >                  self.arrow_patch.figure = self.figure
> >              self.arrow_patch.draw(renderer)
> >
> > -        Text.draw(self, renderer)
> > +        astext = copy.copy(self)
> > +        astext.__class__ = Text
>
> @WeatherGod <https://github.com/WeatherGod> Yes, that is correct (see my
> comment on issue #4139
> <https://github.com/matplotlib/matplotlib/issues/4139>).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/4145/files#r25206783>.
>